x/ibc-hooks/contracts/crosschain-swaps/src/execute.rs

```diff
@@ -0,0 +1,271 @@
+use cosmwasm_std::{coins, from_binary, to_binary, wasm_execute, BankMsg, Reply, Timestamp};
+use cosmwasm_std::{Addr, Coin, DepsMut, Response, SubMsg, SubMsgResponse, SubMsgResult};
+use swaprouter::msg::{ExecuteMsg as SwapRouterExecute, Slippage, SwapResponse};
+
+use crate::checks::{validate_memo, validate_receiver};
+use crate::consts::{CALLBACK_KEY, FORWARD_REPLY_ID, PACKET_LIFETIME, SWAP_REPLY_ID};
+use crate::ibc::{MsgTransfer, MsgTransferResponse};
+use crate::msg::{CrosschainSwapResponse, Recovery};
+
+use crate::state::{
+    ForwardMsgReplyState, ForwardTo, IBCTransfer, Status, SwapMsgReplyState, CONFIG,
+    FORWARD_REPLY_STATES, INFLIGHT_PACKETS, RECOVERY_STATES, SWAP_REPLY_STATES,
+};
+use crate::ContractError;
+
+/// This is the main execute call of this contract.
+///
+/// It's objective is to trigger a swap between the supplied pairs
+///
+#[allow(clippy::too_many_arguments)]
+pub fn swap_and_forward(
+    deps: DepsMut,
+    block_time: Timestamp,
+    contract_addr: Addr,
+    input_coin: Coin,
+    output_denom: String,
+    slippage: Slippage,
```

Comment:
(minor nit, feel free to ignore)

I feel like ideally the aliases for this import would be:
`SwapRouter::Slippage`

Reply:
I think that makes sense

Reply:
changed.